src/GamePool.jl

```diff
@@ -228,26 +220,24 @@ A `target` as a string must have `length(target) == N`.
 See also: [`showgame!`](@ref)
 """
 function playgame!(gp::GamePool{N}, ind::Integer) where {N}
-    psz = length(gp.targetpool)
-    0 < ind ≤ psz || throw(ArgumentError("condition 1 ≤ ind ≤ $psz not satisfied"))
+    gp.validtargets[ind] || throw(ArgumentError("ind = $ind is not a valid target index"))
     reset!(gp)
     maxscore = (3^N) - 1
-    while true
+    while true     # FIXME: there should be a better way of writing this loop
```

Comment:
```suggestion
    while (sc = score(scoreupdate!(gp, sc), ind)) != maxscore)
```
Not sure if that's really more readable (very C programming) ..... but moving the check to the loop condition itself would help

Reply:
Turns out it doesn't quite work because of the order in which the operations need to be done.  The `score` has to be evaluated then the `scoreupdate!`, which also does a test for `sc == maxscore` then the test in this loop.  I will leave it as it is.